### PR TITLE
DRILL-92: Cassandra storage plugin implementation based on Calcite adapter

### PR DESCRIPTION
**Pull request by vvysotskyi**
# [DRILL-92](https://issues.apache.org/jira/browse/DRILL-92): Cassandra storage plugin implementation based on Calcite adapter

## Description
This PR introduces a plugin for Cassandra and Scylla based on Calcite's Cassandra adapter.

Please note, that this PR requires https://github.com/vvysotskyi/drill-calcite/pull/3 to be merged.

## Documentation
Docs on the Drill web site should be updated to state the support of the Cassandra and Scylla plugin.

## Testing
Added unit tests, checked with Cassandra and Scylla.

### CODE REVIEW

File: contrib/storage-cassandra/pom.xml [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    </dependency>
+    <dependency>
+      <groupId>net.hydromatic</groupId>
+      <artifactId>foodmart-data-json</artifactId>
+      <version>0.4</version>
+      <scope>test</scope>
+    </dependency>
+  </dependencies>
+
+</project>
```
Comment (ihuzenko) [inline contrib/storage-cassandra/pom.xml:96]: new line

File: contrib/storage-cassandra/src/main/java/org/apache/drill/exec/store/cassandra/CassandraStoragePlugin.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    return true;
+  }
+
+  @Override
+  public Set<? extends RelOptRule> getOptimizerRules(OptimizerRulesContext optimizerContext, PlannerPhase phase) {
+    switch (phase) {
+      case LOGICAL_PRUNE_AND_JOIN:
+      case LOGICAL_PRUNE:
+      case PARTITION_PRUNING:
+        return Collections.emptySet();
```
Comment (ihuzenko) [inline contrib/storage-cassandra/src/main/java/org/apache/drill/exec/store/cassandra/CassandraStoragePlugin.java:69]: Technically here returned also empty immutable set like for **default** case.

**Review by ihuzenko — APPROVED** [review]
LGTM, +1. Just a few minor comments, thank @vvysotskyi for making the changes.

File: contrib/storage-cassandra/pom.xml (hunk 1 above)
Comment (vvysotskyi) [reply contrib/storage-cassandra/pom.xml]: Thanks, fixed.

File: contrib/storage-cassandra/src/main/java/org/apache/drill/exec/store/cassandra/CassandraStoragePlugin.java (hunk 2 above)
Comment (vvysotskyi) [reply contrib/storage-cassandra/src/main/java/org/apache/drill/exec/store/cassandra/CassandraStoragePlugin.java]: Oh, no, forgot to change it here. Thanks for finding it, combined it with the default case.

**Review by vvysotskyi — COMMENTED** [review]
@ihuzenko, thanks for so quick review!

**Comment by cgivre** [comment]
@vvysotskyi Can we merge this?

**Pull request merged**